src/library-authoring/create-container/CreateContainerModal.test.tsx

```diff
@@ -0,0 +1,241 @@
+  setContainerId: jest.fn(),
+  readOnly: false,
+  isLoadingLibraryData: false,
+  showOnlyPublished: false,
+  extraFilter: [],
+  isCreateCollectionModalOpen: false,
+  openCreateCollectionModal: jest.fn(),
+  closeCreateCollectionModal: jest.fn(),
+  createContainerModalType: ContainerType.Chapter,
+  setCreateContainerModalType: jest.fn(),
+  componentBeingEdited: undefined,
+  openComponentEditor: jest.fn(),
+  closeComponentEditor: jest.fn(),
+  setCollectionId: jest.fn(),
+};
+
+const defaultToastContext: ToastContextData = {
+  showToast: mockShowToast,
+  closeToast: mockCloseToast,
+  toastMessage: null,
+  toastAction: undefined,
+};
+
+const renderComponent = (
+  toastContextValue = defaultToastContext,
+) => {
+  // Set up the mock implementation for useLibraryContext
+  jest.mocked(useLibraryContext).mockReturnValue(defaultLibraryContext);
+
+  return render(
+    <QueryClientProvider client={queryClient}>
+      <IntlProvider locale="en" messages={{}}>
+        <ToastContext.Provider value={toastContextValue}>
+          <CreateContainerModal />
+        </ToastContext.Provider>
+      </IntlProvider>
+    </QueryClientProvider>,
+  );
+};
+
+describe('CreateContainerModal container linking', () => {
+  beforeEach(() => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'test-user',
+        administrator: true,
+        roles: [],
+      },
+    });
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+    jest.clearAllMocks();
+
+    // Reset the useLibraryRoutes mock to default values
+    mockUseLibraryRoutes.mockReturnValue({
+      navigateTo: mockNavigateTo,
+      insideCollection: true,
+      insideSection: false,
+      insideSubsection: false,
+    });
+  });
+
+  it('links container to collection when inside a collection', async () => {
+    // Mock the container creation API
+    axiosMock.onPost(/\/api\/libraries\/.*\/containers/).reply(200, {
+      id: 'lct:org:libId:collection:1',
+      displayName: 'Test Container',
+    });
+
+    // Mock the collection linking API
+    const collectionItemsUrl = getLibraryCollectionItemsApiUrl('test-library', 'test-collection');
+    axiosMock.onPatch(collectionItemsUrl).reply(200);
+
+    renderComponent();
+
+    // Fill in the form
+    const nameInput = screen.getByLabelText(/name/i);
+    await userEvent.type(nameInput, 'Test Container');
+
+    // Submit the form
+    const createButton = screen.getByRole('button', { name: /create/i });
+    await userEvent.click(createButton);
+
+    // Verify API calls
+    await waitFor(() => {
+      // Verify container creation call
+      expect(axiosMock.history.post).toHaveLength(1);
+      expect(axiosMock.history.post[0].url).toMatch(/\/api\/libraries\/.*\/containers/);
+      expect(JSON.parse(axiosMock.history.post[0].data)).toEqual({
+        can_stand_alone: false,
+        container_type: 'section',
+        display_name: 'Test Container',
+      });
```

Comment:
@DanielVZ96 It's recommended to [avoid multiple assertions](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#having-multiple-assertions-in-a-single-waitfor-callback) in a single `waitfor`